ci/brew_formula_updater.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+# Copyright 2024 IBM Corp
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -e
+
+# Validate argument count
+if [ "$#" -ne 4 ]; then
+    echo "Usage: $0 <base_repo_url> <new_version> <brew_tap_repo_url> <formula_file>"
+    exit 1
+fi
+
+# Ensure GitHub token and GH CLI token are set
+if [[ -z "$GITHUB_TOKEN" || -z "$GH_TOKEN" ]]; then
+    echo "Error: Required tokens are not set."
+    exit 1
+fi
+
+BASE_REPO_URL="$1"
+NEW_VERSION="$2"
+BREW_TAP_REPO_URL="$3"
+FORMULA_FILE="$4"
+FORMULA_PATH="Formula/"
+RELEASES_PATH="$BASE_REPO_URL/releases/download"
+
+echo "BASE_REPO_URL: $BASE_REPO_URL"
+echo "NEW_VERSION: $NEW_VERSION"
+echo "BREW_TAP_REPO_URL: $BREW_TAP_REPO_URL"
+echo "FORMULA_FILE: $FORMULA_FILE"
+echo "RELEASES path in base repo - "$RELEASES_PATH
+
+URL_VALUES=(
+    "$RELEASES_PATH/v$NEW_VERSION/pvsadm-darwin-amd64.tar.gz"
+    "$RELEASES_PATH/v$NEW_VERSION/pvsadm-darwin-arm64.tar.gz"
+    "$RELEASES_PATH/v$NEW_VERSION/pvsadm-linux-amd64.tar.gz"
+)
+
+# Compute SHA256 checksum
+compute_sha256() {
+    local url="$1"
+    local temp_file="$(mktemp)"
+    http_status=$(curl -L -w "%{http_code}" -o "$temp_file" "$url")
+    
+    # Check HTTP status 200
+    if [ "$http_status" -ne 200 ]; then
+        echo "Error: Failed to download $url (HTTP status: $http_status)"
+        rm "$temp_file"
+        exit 1
+    fi
+
+    # Verify the content is not HTML
+    if grep -iq "<!doctype\|<html\|<head\|<body" <(head -n 10 "$temp_file"); then
+        echo "Error: Downloaded content is not a valid tar. Possibly an HTML error page."
+        rm "$temp_file"
+        exit 1
+    fi
+
+    local sha256
+    sha256=$(shasum -a 256 "$temp_file" | awk '{print $1}')
+    rm "$temp_file"
+    # Uncomment for testing purposes echo "SHA256 for $url: $sha256"
+    echo "$sha256"
+}
+
+# Compute SHA256 checksums for the new version
+SHAs=()
+for url in "${URL_VALUES[@]}"; do
+    SHA=$(compute_sha256 "$url")
+    if [ -z "$SHA" ]; then
+        echo "Error: SHA256 could not be computed for $url"
+        exit 1
+    fi
+    SHAs+=("$SHA")
+    echo $SHA
+    sleep 1
+done
+
+# Clone the repository having brew formula and checkout a new branch for pushing updates
+git clone "$BREW_TAP_REPO_URL" brew_tap_repo_temp
+if [ $? -ne 0 ]; then
+    echo "Error: failed to clone the repository"
+    exit 1
+fi
+
+cd brew_tap_repo_temp || { echo "Error: Failed to navigate to brew_tap_repo_temp"; exit 1; }
```

Comment:
use mktemp command to generate the temporary folder and assign it to variable and use it everywhere.

Reply:
Done